### ISSUE TITLE
Relist pricing with "same relist $" unchecked

### ISSUE THREAD

**Issue by shuohan27** [hunterjm/futgui#21]
Instead of relisting unsold cards for the current rate, the program will relist for a significant amount below the current "bid up to" amount.  For example, after the 1 hour runs out for a player listed at BIN 27500, I get a manually relisting message, I get no update on the list of for sale/bidded on players in the area above the console, then a few minutes later I get a message that the player sold for 13500.  The updated BIN in this case should have been 29000.  I've noticed the program losing coins this way several times now.  I'll try with the "same relist $" checked as a workaround for now.

**Comment by aftershock9** [comment]
I think (from my testing), it checks the lowest BIN on futbin and sometimes futbin is not a real time reflection on the market. I agree with you and that the update price should be based on the real time reflection of the market. Similar in the way that the players price is initially set. 

Hunter, is this possible to  set the players price based on real time market when the "same relist $" is unchecked?

**Comment by Faruk-S** [comment]
I have same problem. Alves sold for 3600 :)

**Comment by hunterjm** [comment]
As of 0.7.0, It doesn't use FUTBIN anymore.  It uses a custom lowest BIN finder.  Are you guys sure that the lowest BIN of the player didn't drop sometime during the night?  I might change it to look at the second lowest BIN to stop outliers like these from happening.

Otherwise, the "Same Relist $" feature will make sure that cards bought for a certain price will always be listed for the initial list price instead of moving with the market.  That's why it's there ;)

**Comment by shuohan27** [comment]
Jason, I watched the current pricing list on the left based on its polling once per hour, and it always stayed within several thousand and looked like the correct price.  From using the updated price polling and manually checking on the webapp, it looks like that portion works great.  I think it's best to keep that as it is, we need to know the lowest BIN even if it's a single outlier.  If you decide to put in much more complex logic (ie: card A is 10% cheaper than all others) then that might be a different story, but also unnecessarily complex.

As far as the issue I'm reporting, when "Same Relist $" is unchecked it definitely sometimes doesn't use the currently polled BIN pricing.  In another instance yesterday I saw it had 3 players listed, starting higher than current BIN and reducing around 5-10k per hour until they sold.  This problem is obviously a little tricky to recreate since it doesn't really get logged in the bidding/listing log, only showing what a card was purchased for and what it sold for (several hours later), but the issue definitely is there.

Thanks again for your attention to this!

**Comment by shuohan27** [comment]
For what it's worth, the cards I target are in the 20k-30k range.  It initially lists them at the correct BIN price, then all three were hiked to 31500 (above the current BIN) and were reduced roughly 5-10k each hour until they sold, not apparently referring to the price list.  Oddly, these three different players with different current prices were listed for the same price with the same reductions in price each hour.

**Comment by hunterjm** [comment]
The issue I have is there is no logic in the code to do what you describe.  Unless the market is shifting that far down for lowest BIN, everything should stay the same.  See below for better commented code that handles relisting players (located in `core/bid.py` starting at line `259`).

``` python
# relist items
expired = sum([i['tradeState'] == 'expired' for i in tradepile])
if expired > 0:

    relistFailed = False
    # If relist at same price is set in settings, use the "relist all" functionality
    if settings['relistAll']:
        try:
            api.relist()
        except InternalServerError:
            relistFailed = True
            pass

    # If we want to change the price, or the "relist all" failed
    if not settings['relistAll'] or relistFailed:
        q.put('%s    Manually re-listing %d players.\n' % (time.strftime('%Y-%m-%d %H:%M:%S'), expired))
        for i in tradepile:
            baseId = str(abs(item['resourceId'] + 0x80000000))
            if baseId in bidDetails:
                # Set the sell and bin prices to either previous price, or current price based on "relist for same $" setting
                sell = i['startingBid'] if settings['relistAll'] else bidDetails[baseId]['sell']
                binPrice = i['buyNowPrice'] if settings['relistAll'] else bidDetails[baseId]['binPrice']
                if i['tradeState'] == 'expired' and sell and binPrice:
                    api.sell(i['id'], sell, binPrice)
```

`i['startingBid']` and `i['buyNowPrice']` are the previous list and BIN prices.  They get used if you have "Same Relist $" checked.  Otherwise, it uses `bidDetails[baseId]['sell']` and `bidDetails[baseId]['binPrice']` which is the current calculated price that shows on the left.  This has to mean that at some point, the lowest BIN dropped to that price.

Next time, can you copy the entire contents of the log window to a text file and upload it here so I can see exactly what happened?

**Comment by shuohan27** [comment]
See the attached console log from today.  There are at least a couple players in here that were sold for too cheap (ozil at 14:36:41 and courtois at 14:45:08).
[log.txt](https://github.com/hunterjm/futgui/files/50131/log.txt)

**Comment by hunterjm** [comment]
I found the bug, but won't be able to release a fix until I get off work tonight.

``` python
for i in tradepile:
    baseId = str(abs(item['resourceId'] + 0x80000000))
```

should be

``` python
for i in tradepile:
    baseId = str(abs(i['resourceId'] + 0x80000000))
```

**Comment by hunterjm** [comment]
Fixed in [v0.7.1-beta](https://github.com/hunterjm/futgui/releases/tag/v0.7.1-beta)
